### DISCUSSION TITLE
Command can not be found in CLI

### DISCUSSION THREAD

**Discussion by smile0821** [apache/airflow#34273] [Q&A] [answered]
1. I'd like to mark several tasks as failed by passing dag_id, run_id and task_ids. 
2. I'd like to mark several tasks as success by passing dag_id, run_id and task_ids.
3. I'd like to implement the function of clear in the airflow UI by passing dag_id, run_id and several task_ids(task_ids are optional). 
Are there corresponding commands in the airflow CLI? If yes, please give me corresponding example commands, thanks

**Comment by Taragolis** [comment, marked as answer]
1 and 2: Not exists in cli, exists in REST API, see endpoints:
- [dags/{dag_id}/updateTaskInstancesState](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_set_task_instances_state) 
- [dags/{dag_id}/dagRuns/{dag_run_id}/taskInstances/{task_id}](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/patch_task_instance) 
- [dags/{dag_id}/dagRuns/{dag_run_id}/taskInstances/{task_id}/{map_index}](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/patch_mapped_task_instance)

3: Exists in cli [`airflow tasks clear`](https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#clear) and also exists in REST API, see enpoint:
- [dags/{dag_id}/clearTaskInstances](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_clear_task_instances)

**Reply by smile0821** [reply]
Hi Taragolis, for airflow tasks clear in cli, I think it can clear tasks if I offer task regex and dag_id, but it can't clear specific tasks when I offer task_id+run_id+dag_id, which means airflow cli just offers several simple functions, and REST API has similar functions with UI, thanks for your reply, have a nice day~
